### PR TITLE
Added Debug Plugin

### LINKED ISSUE #6: Create Baseline Debug Plugin
Create Baseline Debug Plugin for developer testing of new features. Enables printing to terminal of entity values such as position.

### PR DESCRIPTION
**Pull request by joshuawalker3**
Created a baseline debug plugin that can be used for printing debug info to the command line. Systems can be added to the plugin as needed and debug can be disabled by removing the plugin in main. Tested using system that printed message to terminal every frame

[//]: # (Please delete any unused sections before submitting)

# Objective
- A baseline debugging plugin was needed to provide debugging functionality using the terminal.
- Fixes #6 

## Solution
Implemented DebugPlugin component.

## New components
DebugPlugin was created

## New systems
added print_test system to DebugPlugin component to provide baseline known good function for testing purposes

## Testing
Tested by running app and verified that the correct message was printed to the terminal every frame

### CODE REVIEW

File: game/src/debug/debug_plugin.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+use bevy::prelude::*;
+
+pub struct DebugPlugin;
```
Comment (Remi-Godin) [inline game/src/debug/debug_plugin.rs:3]: Please add documentation. You can do so by using the triple backslash notation:
```Rust
/// This describes what this plugin does
/// Maybe it also describe how to use it and expand on it
pub struct DebugPlugin;
```
[More info on rust documentation here](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html)

File: game/src/debug/debug_plugin.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+//add the system(s) you wish to use to see debug info
+impl Plugin for DebugPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(Update, print_test);
```
Comment (Remi-Godin) [inline game/src/debug/debug_plugin.rs:8]: "Debugger active" message should only show in the `Startup` schedule.

File: game/src/debug/debug_plugin.rs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    }
+}
+
+//Function to test if debugger is added correctly
```
Comment (Remi-Godin) [inline game/src/debug/debug_plugin.rs:12]: Like above, use doc strings here to bind your comment to the function.

File: game/src/main.rs [hunk 4]
```diff
@@ -11,5 +13,8 @@ fn main() {
         .add_plugins(PhysicsPlugins::default())
         .add_plugins(LdtkPlugin)
         .add_plugins(CharacterControllerPlugin)
+        //user plugins
+        //debug plugin
```
Comment (Remi-Godin) [inline game/src/main.rs:17]: You can remove the "debug plugin" comment. The code is self explanatory.

File: game/src/debug/debug_plugin.rs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+fn print_test() {
+    info!("The debugger is working!");
```
Comment (Remi-Godin) [inline game/src/debug/debug_plugin.rs:14]: You should rename this function to something like "print_debug_active_message" or something like that. Even if the goal was just to test the system, I think having a message to indicate the debugger is running at the beginning of each run could be beneficial.

**Review by Remi-Godin — CHANGES_REQUESTED** [review]
Please address the comments I left in the review, and then we should be able to merge the PR. Thanks!

File: game/src/debug/debug_plugin.rs (hunk 1 above)
Comment (joshuawalker3) [reply game/src/debug/debug_plugin.rs:3]: Updated

File: game/src/debug/debug_plugin.rs (hunk 2 above)
Comment (joshuawalker3) [reply game/src/debug/debug_plugin.rs:8]: Updated

File: game/src/debug/debug_plugin.rs (hunk 3 above)
Comment (joshuawalker3) [reply game/src/debug/debug_plugin.rs:12]: updated

File: game/src/main.rs (hunk 4 above)
Comment (joshuawalker3) [reply game/src/main.rs:17]: Updated

File: game/src/debug/debug_plugin.rs (hunk 5 above)
Comment (joshuawalker3) [reply game/src/debug/debug_plugin.rs:14]: Updated

**Review by Remi-Godin — APPROVED** [review]

**Pull request merged**